CDN/telemetry_upload.py

```diff
@@ -0,0 +1,224 @@
+    return (r.status_code, es_resp_status, es_resp_err, es_resp_errmsg)
+
+
+# Generate log only for 100th error from timer 
+log_optimization_limit = 100
+log_current_value = 0 
+
+def telemetry_upload_dir():
+    #
+    # Check if telemetry file avalable to sync
+    # If not, just return
+    #
+    global tmDir
+    tm_dir = tmDir
+
+    try:
+        try:
+            os.chdir(tm_dir)
+        except:
+            err_msg = "Directory read error:  " + tm_dir
+            raise BreakoutException
+
+        tmfile_list = os.listdir(tm_dir)
+        #tmfile_list = sorted(os.listdir(tm_dir),key=os.path.getctime);
+        if not tmfile_list:
+            err_msg = "No file in " + tm_dir + " to upload..."
+            raise BreakoutException
+
+        #log.info(' '.join(str(x) for x in tmfile_list))
+        #
+        # We have some files to upload; check net connectivity now
+        # If not connected, just return
+        #
+        netstatus = check_netconnectivity()
+        if not netstatus:
+            err_msg = "Not connected to network..."
+            raise BreakoutException
+        else:
+            log.info("Connected to network...")
+   
+        tmfile_timesorted_list = sorted(tmfile_list, key=os.path.getmtime)
+        #log.info(' '.join(str(x) for x in tmfile_list))
+
+        #
+        # Upload the first file with existing credential
+        # If we get unauthorized/rate limited error
+        # Handle that
+        #
+        ratelimit_count = 100
+        for filename in tmfile_timesorted_list:
+            status, es_resp_status, es_resp_err, es_resp_errmsg = telemetry_upload_file(filename, tm_jwt, tmURL)
+            if es_resp_status == "successful" or es_resp_err == "INVALID_DATA_ERROR":
+                log.info("telemetry upload(%s) status: %s %s" % 
+                        (filename, es_resp_status, es_resp_errmsg))
+                # delete this file
+                os.remove(filename)
+            else:
+                log.info("telemetry upload(%s) status: %d es_status: %s es_err: %s es_errmsg: %s" %
+                        (filename, status, es_resp_status, es_resp_err, es_resp_errmsg))
+                # TODO: If failed due to token expiry, we need to regenerate the token
+                # For now, we are generating the token on every reboot anyway
+
+            # Ensure we are not rate limited by server
+            if ratelimit_count < 1:
+                break
+            else:
+                ratelimit_count = ratelimit_count - 1
+    except:
+        # Don't flood with logs from timer
+        global log_optimization_limit, log_current_value
+        log_current_value  = log_current_value + 1
+
+        if log_current_value == 1:
+            log.error(err_msg)
+        elif log_current_value >= log_optimization_limit:
+            log_current_value  = 0
+            
+    # The below line required for next timer fire
+    threading.Timer(60, telemetry_upload_dir).start()
```

Comment:
IMO, 60 sec is too fast. This needs to be well designed..

Reply:
Any suggestions here?